examples/map_embeddings_progressively.py

```diff
@@ -1,22 +1,24 @@
 from nomic import atlas
 import numpy as np
 
+from nomic import atlas
+import numpy as np
 
 num_embeddings = 1000
-embeddings = np.random.rand(num_embeddings, 256)
-data = [{'id': i} for i in range(len(embeddings))]
-
-first_upload_embeddings = embeddings[:500, :]
-second_upload_embeddings = embeddings[500:, :]
-first_upload_data = data[:500]
-second_upload_data = data[500:]
+embeddings = np.random.rand(num_embeddings, 10)
+data = [{'upload': '1'} for i in range(len(embeddings))]
 
-response = atlas.map_embeddings(embeddings=first_upload_embeddings,
-                                data=first_upload_data,
-                                is_public=True)
+project = atlas.map_embeddings(embeddings=embeddings,
+                               data=data,
+                               map_name='A Map That Gets Updated',
+                               colorable_fields=['upload'])
+map = project.get_map('A Map That Gets Updated')
+print(map)
 
-print('First upload response: ', response)
-response = atlas.update_maps(embeddings=second_upload_embeddings,
-                             data=second_upload_data)
+# embeddings with shifted mean.
+embeddings += np.ones(shape=(num_embeddings, 10))
+data = [{'upload': '2'} for i in range(len(embeddings))]
 
-print('Second upload response: ', response)
+with project.block_until_accepting_data() as project:
+    project.add_embeddings(embeddings=embeddings, data=data)
+    project.rebuild_maps()
```

Comment:
I wonder if it would be clearer to have the block method be called `project.wait_for_upload_lock` or something?
`block_until_accepting_data` to me implies that the context manager *creates* a block, rather than waits for an existing one to be released.

Reply:
Didn't really think of it that way. It isnt just an upload block, it also blocks operations such a reindexing, nn search, etc.

I would be open to changing it to wait_for_lock_release() or something akin.

Thoughts?